From bcb3d7533e10b185d9387a8fdea370d306a6bcc2 Mon Sep 17 00:00:00 2001
From: Dmitry Monakhov <dmonakhov@openvz.org>
Date: Sat, 7 Apr 2012 04:15:47 +0000
Subject: [PATCH] openssl: Prepare libjingle specific build

- Enable DTLS support
- Disable all unnecessery targets in order to reduce build time:
   a) disable all dynamic libs, leave only static ssl and crypto wich
      will be statically linked in to dynamyc libjingle
   b) disable all executables
---
 Android.mk        |    1 -
 crypto/Android.mk |   76 +++++++++-------------------------------------------
 ssl/Android.mk    |   43 ++++++++++++------------------
 3 files changed, 30 insertions(+), 90 deletions(-)

diff --git a/Android.mk b/Android.mk
index 741123b..95f9371 100644
--- a/Android.mk
+++ b/Android.mk
@@ -3,7 +3,6 @@ LOCAL_PATH := $(call my-dir)
 subdirs := $(addprefix $(LOCAL_PATH)/,$(addsuffix /Android.mk, \
 		crypto \
 		ssl \
-		apps \
 	))
 
 include $(subdirs)
diff --git a/crypto/Android.mk b/crypto/Android.mk
index 222bc96..9e8d8c4 100644
--- a/crypto/Android.mk
+++ b/crypto/Android.mk
@@ -470,6 +470,8 @@ local_src_files := \
 	x509v3/v3_utl.c \
 	x509v3/v3err.c
 
+local_dtls_src_files := pqueue/pqueue.c
+
 ifeq ($(BOARD_USE_OPENSSL_ENGINE),true)
 LOCAL_SRC_FILES += \
        engine/eng_err.c \
@@ -496,40 +498,15 @@ LOCAL_SRC_FILES += \
 endif
 
 local_c_includes := \
-	external/openssl \
-	external/openssl/crypto/asn1 \
-	external/openssl/crypto/evp \
-	external/openssl/include \
-	external/openssl/include/openssl \
-	external/zlib
+	$(LOCAL_PATH)/.. \
+	$(LOCAL_PATH)/../crypto/asn1 \
+	$(LOCAL_PATH)/../crypto/evp \
+	$(LOCAL_PATH)/../include \
+	$(LOCAL_PATH)/../include/openssl \
 
-local_c_flags := -DNO_WINDOWS_BRAINDEATH
+#	$(LOCAL_PATH)/../zlib
 
-#######################################
-
-# target
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../android-config.mk
-LOCAL_SRC_FILES += $(local_src_files)
-LOCAL_CFLAGS += $(local_c_flags)
-LOCAL_C_INCLUDES += $(local_c_includes)
-LOCAL_SHARED_LIBRARIES += libz
-ifeq ($(TARGET_ARCH),arm)
-	LOCAL_SRC_FILES += $(arm_src_files)
-	LOCAL_CFLAGS += $(arm_cflags)
-else
-	LOCAL_SRC_FILES += $(non_arm_src_files)
-endif
-ifeq ($(TARGET_SIMULATOR),true)
-	# Make valgrind happy.
-	LOCAL_CFLAGS += -DPURIFY
-    LOCAL_LDLIBS += -ldl
-else
-	LOCAL_SHARED_LIBRARIES += libdl
-endif
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE:= libcrypto
-include $(BUILD_SHARED_LIBRARY)
+local_c_flags := -DNO_WINDOWS_BRAINDEATH
 
 #######################################
 
@@ -553,37 +530,10 @@ ifeq ($(TARGET_SIMULATOR),true)
 else
 	LOCAL_SHARED_LIBRARIES += libdl
 endif
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE:= libcrypto
-include $(BUILD_STATIC_LIBRARY)
-
-#######################################
-# host shared library
-ifeq ($(WITH_HOST_DALVIK),true)
-    include $(CLEAR_VARS)
-    include $(LOCAL_PATH)/../android-config.mk
-    LOCAL_SRC_FILES += $(local_src_files)
-    LOCAL_CFLAGS += $(local_c_flags) -DPURIFY
-    LOCAL_C_INCLUDES += $(local_c_includes)
-    LOCAL_SRC_FILES += $(non_arm_src_files)
-    LOCAL_STATIC_LIBRARIES += libz
-    LOCAL_LDLIBS += -ldl
-    LOCAL_MODULE_TAGS := optional
-    LOCAL_MODULE:= libcrypto
-    include $(BUILD_HOST_SHARED_LIBRARY)
+ifeq ($(MY_USE_DTLS),1)
+	LOCAL_SRC_FILES += $(local_dtls_src_files)
 endif
-
-########################################
-# host static library, which is used by some SDK tools.
-
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../android-config.mk
-LOCAL_SRC_FILES += $(local_src_files)
-LOCAL_CFLAGS += $(local_c_flags) -DPURIFY
-LOCAL_C_INCLUDES += $(local_c_includes)
-LOCAL_SRC_FILES += $(non_arm_src_files)
-LOCAL_STATIC_LIBRARIES += libz
-LOCAL_LDLIBS += -ldl
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE:= libcrypto_static
-include $(BUILD_HOST_STATIC_LIBRARY)
+include $(BUILD_STATIC_LIBRARY)
+
diff --git a/ssl/Android.mk b/ssl/Android.mk
index fb9d34c..c23582e 100644
--- a/ssl/Android.mk
+++ b/ssl/Android.mk
@@ -1,9 +1,9 @@
 LOCAL_PATH:= $(call my-dir)
 
 local_c_includes := \
-	external/openssl \
-	external/openssl/include \
-	external/openssl/crypto
+	$(LOCAL_PATH)/.. \
+	$(LOCAL_PATH)/../include \
+	$(LOCAL_PATH)/../crypto
 
 local_src_files:= \
 	s2_meth.c \
@@ -44,32 +44,23 @@ local_src_files:= \
 	ssl_err.c \
 	kssl.c
 
+local_dtls_src_files := \
+	d1_meth.c \
+	d1_srvr.c \
+	d1_clnt.c \
+	d1_lib.c \
+	d1_pkt.c \
+	d1_both.c \
+	d1_enc.c
+
 include $(CLEAR_VARS)
 include $(LOCAL_PATH)/../android-config.mk
 LOCAL_SRC_FILES += $(local_src_files)
-LOCAL_C_INCLUDES += $(local_c_includes)
-LOCAL_SHARED_LIBRARIES += libcrypto
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE:= libssl
-include $(BUILD_SHARED_LIBRARY)
-
-ifeq ($(WITH_HOST_DALVIK),true)
-    include $(CLEAR_VARS)
-    include $(LOCAL_PATH)/../android-config.mk
-    LOCAL_SRC_FILES += $(local_src_files)
-    LOCAL_C_INCLUDES += $(local_c_includes)
-    LOCAL_SHARED_LIBRARIES += libcrypto
-    LOCAL_MODULE_TAGS := optional
-    LOCAL_MODULE:= libssl
-    include $(BUILD_HOST_SHARED_LIBRARY)
+ifeq ($(MY_USE_DTLS),1)
+	LOCAL_SRC_FILES += $(local_dtls_src_files)
 endif
-
-# ssltest
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../android-config.mk
-LOCAL_SRC_FILES:= ssltest.c
 LOCAL_C_INCLUDES += $(local_c_includes)
-LOCAL_SHARED_LIBRARIES := libssl libcrypto
-LOCAL_MODULE:= ssltest
+LOCAL_STATIC_LIBRARIES += libcrypto_static
 LOCAL_MODULE_TAGS := optional
-include $(BUILD_EXECUTABLE)
+LOCAL_MODULE:= libssl_static
+include $(BUILD_STATIC_LIBRARY)
-- 
1.7.5.4

